py3status/modules/battery_level.py

```diff
@@ -125,13 +111,88 @@ def on_click(self, i3s_output_list, i3s_config, event):
                 stderr=open('/dev/null', 'w')
             )
 
+    def _provide_backwards_compatibility(self):
+        # Backwards compatibility for 'mode' parameter
+        if self.format == FORMAT and self.mode == 'ascii_bar':
+            self.format = "{ascii_bar}"
+        if self.format == FORMAT and self.mode == 'text':
+            self.format = "Battery: {percent}"
+
+        # Backwards compatibility for 'show_percent_with_blocks' parameter
+        if self.format == FORMAT and self.show_percent_with_blocks:
+            self.format = "{icon} {percent}%"
+
+        # Backwards compatibility for '{}' option in format string
+        self.format = self.format.replace('{}', '{percent}')
+
+    def _refresh_battery_info(self):
+        # Example acpi raw output: "Battery 0: Discharging, 43%, 00:59:20 remaining"
+        acpi_raw = subprocess.check_output(["acpi"], stderr=subprocess.STDOUT)
+        acpi_unicode = acpi_raw.decode("UTF-8")
+
+        #  Example list: ['Battery', '0:', 'Discharging', '43%', '00:59:20', 'remaining']
+        self.acpi_list = acpi_unicode.split(' ')
+
+        self.charging = self.acpi_list[2][:8] == "Charging"
+        self.percent_charged = int(self.acpi_list[3][:-2])
+
+    def _update_ascii_bar(self):
+        self.ascii_bar = FULL_BLOCK * int(self.percent_charged/10)
+        if self.charging:
+            self.ascii_bar += EMPTY_BLOCK_CHARGING * (10 - int(self.percent_charged/10))
+        else:
+            self.ascii_bar += EMPTY_BLOCK_DISCHARGING * (10 - int(self.percent_charged/10))
+
+    def _update_icon(self):
+        if self.charging:
+            self.icon = self.charging_character
+        else:
+            self.icon = self.blocks[int(math.ceil(self.percent_charged/100*(len(self.blocks) - 1)))]
+
+    def _update_full_text(self):
+        self.full_text = self.format                                          \
```

Comment:
let's be pythonic till the end, please use `self.full_text = self.format.format(ascii_bar=self.ascii_bar, icon=self.icon, percent=self.percent_charged)` ? :)